homeassistant/components/esphome/__init__.py

```diff
@@ -129,6 +129,16 @@ def async_on_service_call(service: HomeassistantServiceCall) -> None:
                     "Can only generate events under esphome domain! (%s)", host
                 )
                 return
+
+            # Call native tag scan
+            if service_name == "tag_scanned":
+                tag_id = service_data["tag_id"]
+                device_id = service_data["device_id"]
+                hass.async_create_task(
+                    hass.components.tag.async_scan_tag(tag_id, device_id)
```

Comment:
`device_id` needs to be the device registry ID of ESPHome device in Home Assistant. 

The device registry entry is returned from calling `device_registry.async_get_or_create` on line 278. The ID is available as `entry.id`

Reply:
#40132